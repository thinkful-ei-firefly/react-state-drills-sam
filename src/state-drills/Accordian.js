import React from 'react';


//prop.sections = [{
//  title: string
//  content: string
// }]
class Accordian extends React.Component {

  state = {
    openItem: null
  }

  handleClick = (itemKey) => {
    this.setState({openItem: itemKey})
  }

  render() {
    return (
      <div className='accordian'>
        <ul>
          {this.props.sections && this.props.sections.map((item, index) => { return (
            <li key={index}>
              <button onClick={()=>this.handleClick(index)}>{item.title}</button>
              <p>{this.state.openItem===index && item.content}</p>
            </li>
          )})}
        </ul>
      </div>
    );
  }
}

export default Accordian;