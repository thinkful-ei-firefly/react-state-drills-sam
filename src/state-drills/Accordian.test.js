import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordian from './Accordian';
import store from './Accordian.store.js';


describe ('Accordian testing', () => {
  it ('renders without crashing', () => {
    const div=document.createElement('div');
    ReactDOM.render(<Accordian sections={store}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it ('renders this UI as expected', () => {
    const tree=renderer.create(<Accordian sections={store}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it ('renders nothing with no props', () => {
    const tree=renderer.create(<Accordian />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});