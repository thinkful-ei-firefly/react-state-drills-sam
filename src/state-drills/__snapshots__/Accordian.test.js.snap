// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian testing renders nothing with no props 1`] = `
<div
  className="accordian"
>
  <ul />
</div>
`;

exports[`Accordian testing renders this UI as expected 1`] = `
<div
  className="accordian"
>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p />
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;
